opium-testing/src/opium_testing.ml

```diff
@@ -0,0 +1,114 @@
+module Testable = struct
+  let status = Alcotest.of_pp Opium_kernel.Status.pp_hum
+  let meth = Alcotest.of_pp Opium_kernel.Method.pp_hum
+  let version = Alcotest.of_pp Opium_kernel.Version.pp_hum
+  let body = Alcotest.of_pp Opium_kernel.Body.pp_hum
+  let request = Alcotest.of_pp Opium_kernel.Request.pp_hum
+  let response = Alcotest.of_pp Opium_kernel.Response.pp_hum
+end
+
+let handle_request app =
```

Comment:
Maybe be worth it to split up `Server_connection.run` instead of duplicating the logic here.